PopulateTools/decidim-module-extra_user_fields
# [Feature] - Allow admin to enable / disable extra user fields

#### Description

This PR implements the possibility for admins to enable / disable extra user fields in the signup form. Also, admins can choose which field available to add to the registration form.

When the extra user field is enabled, and at least one field is selected, a new section will be add to the registration form and account form.

#### Related to

Issue #8, see issue for screenshots

#### How to test

1. Add the migration to the application (it creates a new column `extra_user_fields` to the organizations table)
2. Log in as admin, and go to Settings > Manage extra user fields
3. Enable the extra user field and at least one field
4. Go to registration form, see the fields
5. Go to account form, see the new fields

#### Information

* Exports does not take in account if fields are enabled or not
* extra_user_fields column is a jsonb with this format
```
{
  "enabled" => Boolean,
  "custom_field" => { "enabled" => Boolean }
}
```

The choice of Hash for field in the jsonb column seems to me easier for future needs.

#### Tasks

- [x] Create new migration
- [x] Add configuration form in BO
- [x] Toggle fields in registration form / account form
- [x] Add inclusion validation for gender field
- [x] Add admin permissions for new controller
- [ ] Add documentation for creating a new field => I am thinking about adding documentation in another PR, because there is a lot of steps I think having a task for create skeleton of new field can be interesting ?

#### Additionnal information

Feel free to suggest or ask for changes.

Thanks
